pontoon/base/middleware.py

```diff
@@ -38,6 +42,19 @@ def process_request(self, request):
         if ip in settings.BLOCKED_IPS:
             return HttpResponseForbidden("<h1>Forbidden</h1>")
 
+        # Convert request IP string to an IP object, check if it belongs to an
+        # IP range
+        try:
+            ip_obj = ip_address(ip)
+        except ValueError:
+            log = logging.getLogger(__name__)
+            log.error(f"Invalid IP detected in BlockedIpMiddleware: {ip}")
```

Comment:
Why do we want to log this?

In case an invalid IP address is reported, what can we do about it?

Reply:
I think there is value in knowing that the resulting IP is not valid, and we might need to update the code for edge cases. Right now, we wouldn't even know they exist.

```
        try:
            ip = request.META["HTTP_X_FORWARDED_FOR"]
            # If comma-separated list of IPs, take just the last one
            # http://stackoverflow.com/a/18517550
            ip = ip.split(",")[-1]
        except KeyError:
            ip = request.META["REMOTE_ADDR"]
```